analyses/org.eclipse.tracecompass.incubator.xaf.ui/src/org/eclipse/tracecompass/incubator/internal/xaf/core/statemachine/variable/StateMachineVariableAnalysis.java

```diff
@@ -1310,12 +1307,19 @@ private List<InterruptionDuration> getCriticalPathSteps(InstanceStepInformation
             workerFound = true;
 
             // Compute the path for the period we're interested into
-            TmfVertex vstart = graph.getVertexAt(Objects.requireNonNull(ti.getStartTime()), lw);
-            TmfVertex vend = graph.getVertexAt(Objects.requireNonNull(ti.getEndTime()), lw);
-            TmfGraph path = new CriticalPathAlgorithmBounded(graph).compute(Objects.requireNonNull(vstart), Objects.requireNonNull(vend));
+//            ITmfVertex vstart = graph.getVertexAt(Objects.requireNonNull(ti.getStartTime()), lw);
+//            ITmfVertex vend = graph.getVertexAt(Objects.requireNonNull(ti.getEndTime()), lw);
+
+            // Note: compute() method was removed because it was deprecated. More over when it was deprecated the method implementation
+            // was changed to throw an UnsupportedOperationException().
+            // See https://git.eclipse.org/r/c/tracecompass/org.eclipse.tracecompass/+/182723
+            // So, the implementation was broken and this change just makes it compile
+
+//            TmfGraph path = new CriticalPathAlgorithmBounded(graph).compute(Objects.requireNonNull(vstart), Objects.requireNonNull(vend));
+            ITmfGraph path = graph;
```

Comment:
This should be fixed using the OSCriticalPathAlgorithm instead. I think it should be one line:
` TmfGraph path = new OSCriticalPathAlgorithmBounded(graph).compute(Objects.requireNonNull(vstart), Objects.requireNonNull(vend));`

Reply:
I think I figured it out